docs/node/development/contributing.md

```diff
@@ -1,168 +1,55 @@
+
+As the codebase and these guidelines evolve, the codebase may not always be in full compliance with the guidelines.
+Existing code should not be taken as and example for new code.
+PRs that update the style of existing code should be made separate from PRs that change or add code.
+
+
+## VILLASframework Coding Guidelines
+1. Automated formatting tools
+    1. **clang-format:** The VILLASframework uses clang-format to enforce a consistent code style. Please refer to the [clang-format configuration](https://github.com/VILLASframework/node/blob/master/.clang-format). Please run `clang-format` on your code before making a commit.
+    2. **EditorConfig:** Please refer to the [editorconfig file](https://github.com/VILLASframework/node/blob/master/.editorconfig) on how to setup your editor.
+2. Naming conventions
+    1. Generally, we prefer verbose names over short names, especially if they are outside facing. Avoid naming things similarly.
+    2. **File naming:** We use snake case for files. We use .cpp and .hpp as C++ file extensions.
+    3. **Variables, functions, and types:** We use camel case for variables, functions, and types. Variables and functions begin with a lower case letter. Types begin with a capital case letter. Defines and Macros are all capital case, but `constexpr` variables should be prefered over defines.
+    4. **config files:** We use snake case in JSON and config files. The JSON schemas should be kept up to date.
+3. Comments
+    1. A comment describing what a file contains should be part of every header file. If the file contains a class, this should include a describtion of its purpose. This description should not be duplicated in the source file.
+    2. A SPDX license header must be present in every file. This mentions the authors, a copyright notice and a reference to the Apache-2.0 license. An example is:
+    ```cpp
+    /* This is a short description of what this file does.
+    *
+    * More detailed description expanding on the previous one.
+    * This can also containt information about how to use the code,
+    * some design decisions or other information that is not
+    * obvious from the code.
+    *
+    * Author: FirstName LastName <email@domain.tld>
+    * Author: FirstName2 LastName2 <email2@domain.tld>
+    * SPDX-FileCopyrightText: 2024 FirstName LastName <email@domain.tld>
+    * SPDX-FileCopyrightText: 2022-2023 Company of Author2
+    * SPDX-License-Identifier: Apache-2.0
+    */
+    ```
+    3. Commented code that is not part of an explanation is not allowed.
+4. Commits
+    - We use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
+    - You must agree to the [DCO](https://developercertificate.org/) and sign-off your commits. This is also enforced by the CI.
+5. C++ Features
+    1. **Error Handling:** We do not use exceptions in new code. Instead use `std::expected` to return errors as optionals. 
```

Comment:
Can we omit this part for now? I would like to gather some experience before committing to this. Especially the requirement of C++23 is currently a bit difficult in my use case..

The OPAL-RT Linux currently does not ship with a C++23 compiler and I would like to avoid introducing C++23 features for now which can not be polyfilled.

Reply:
OK I will work on a separate PR trying to introduce this while being backward compatible. Is C++17 as a requirement ok, or do you need an earlier version?